### PR TITLE
Documentation for Create Auxiliary Image feature

### PR DESCRIPTION
**Pull request by ddsharpe**
Update documentation version and project version to 1.10.0 for the release of `createAuxImage`.

### CODE REVIEW

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+title: "Create Auxiliary Image"
+date: 2019-02-23
+draft: false
+weight: 2
+description: "The createAuxImage command creates a new container image with WDT models, archives, and variable files."
+---
+
+
+The `createAuxImage` command helps build a container image from a given base OS image. 
+The required option for the command is marked. There are a number of optional parameters for the create feature.
```
Comment (rosemarymarano) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:11]: Just as a suggestion, I would reverse the order of these two sentences: There are a number of optional parameters for the create feature. The required option for the command is marked.

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+| `--chown` | `userid:groupid` for JDK/Middleware installs and patches.  | `oracle:oracle` |
+| `--dryRun` | Skip Docker build execution and print the Dockerfile to stdout.  |  |
+| `--fromImage` | Container image to use as a base image when creating a new image. | `busybox`  |
+| `--httpProxyUrl` | Proxy for the HTTP protocol. Example: `http://myproxy:80` or `http:user:passwd@myproxy:8080`  |   |
+| `--httpsProxyUrl` | Proxy for the HTTPS protocol. Example: `https://myproxy:80` or `https:user:passwd@myproxy:8080`  |   |
+| `--packageManager` | Override the default package manager for the base image's operating system. Supported values: `APK`, `APTGET`, `NONE`, `YUM`, `ZYPPER`  |   |
+| `--pull` | Always attempt to pull a newer version of base images during the build.  |   |
+| `--tag` | (Required) Tag for the final build image. Example: `store/oracle/mydomain:1`  |   |
+| `--target` | Select the target environment in which the created image will be used. Supported values: `Default` (Docker/Kubernetes), `OpenShift` | `Default`  |
+| `--wdtArchive` | A WDT archive zip file or comma-separated list of files.  |   |
```
Comment (rosemarymarano) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:30]: zip  -> ZIP

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Use it on the command line, as follows:
+
+```bash
+$ imagetool @/path/to/build_args
+```
+
+
+### Usage scenarios
+
+The commands below assume that all the required WDT installer has been downloaded and added to the cache.
```
Comment (rosemarymarano) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:73]: The commands below -> The following commands 
assume that all the required WDT installer has been downloaded -> (this doesn't make sense to me, maybe) assume that the required WDT installer has been downloaded

**Review by rosemarymarano — COMMENTED** [review]
Few edits/one comment

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+---
+title: "Create Auxiliary Image"
+date: 2019-02-23
+draft: false
+weight: 2
+description: "The createAuxImage command creates a new container image with WDT models, archives, and variable files."
```
Comment (tbarnes-us) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:6]: with a WDT install and optional WDT model, archive, or variable files.

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+---
+
+
+The `createAuxImage` command helps build a container image from a given base OS image.
```
Comment (tbarnes-us) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:10]: Add sentences:

The main purpose of this command is for creating images that don't require a Java or WebLogic install, but do require having a WDT install plus optional WDT model, archive, or variable files. This is particular useful for supplying [WebLogic Kubernetes Operator Model-in-Image auxiliary images](https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/model-in-image/auxiliary-images/).

**Review by rjeberhard — APPROVED** [review]

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+| `--fromImage` | Container image to use as a base image when creating a new image. | `busybox`  |
+| `--httpProxyUrl` | Proxy for the HTTP protocol. Example: `http://myproxy:80` or `http:user:passwd@myproxy:8080`  |   |
+| `--httpsProxyUrl` | Proxy for the HTTPS protocol. Example: `https://myproxy:80` or `https:user:passwd@myproxy:8080`  |   |
+| `--packageManager` | Override the default package manager for the base image's operating system. Supported values: `APK`, `APTGET`, `NONE`, `YUM`, `ZYPPER`  |   |
+| `--pull` | Always attempt to pull a newer version of base images during the build.  |   |
+| `--skipCleanup` | Do not delete the build context folder, intermediate images, and failed build containers. For debugging purposes.  |   |
+| `--tag` | (Required) Tag for the final build image. Example: `store/oracle/mydomain:1`  |   |
+| `--target` | Select the target environment in which the created image will be used. Supported values: `Default` (Docker/Kubernetes), `OpenShift` | `Default`  |
+| `--wdtArchive` | A WDT archive ZIP file or comma-separated list of files.  |   |
+| `--wdtHome` | The target folder in the image for the WDT install and models.  | `/auxiliary`  |
```
Comment (tbarnes-us) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:32]: Consider adding sentence:  The WDT install will placed in subdirectory `./weblogic-deploy`.

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+| `--wdtModel` | A WDT model file or a comma-separated list of files.  |   |
+| `--wdtModelHome` | The target location in the image to copy WDT model, variable, and archive files. | `{wdtHome}/models` |
+| `--wdtVariables` | A WDT variables file or comma-separated list of files.  |   |
+| `--wdtVersion` | WDT version to use.  | `latest`  |
```
Comment (tbarnes-us) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:36]: WDT version to use --> WDT version to install in `{wdtHome}/weblogic-deploy`.

Does this support a `none` option? It's conceivable MII aux image users won't need a WDT install - it could already be available in a different aux image...

File: documentation/1.10/content/userguide/tools/create-aux-image.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+### Additional information
+
+#### `--target`
+
+By default, the generated WLS domain in your image will use the best practices defined by Oracle WebLogic Server.
```
Comment (tbarnes-us) [inline documentation/1.10/content/userguide/tools/create-aux-image.md:42]: There's no 'generated WLS domain' in this case, right?  So maybe reword to remove any mention of 'generated WLS domain'?

**Review by jshum2479 — APPROVED** [review]

File: documentation/1.10/content/userguide/tools/create-aux-image.md (hunk 7 above)
Comment (mriccell) [reply documentation/1.10/content/userguide/tools/create-aux-image.md:36]: We can not assume that the WLS binary mage has WDT, most likely it will not have it.  WLS images in OCR do not have WDT.  Aux images are only for the  MII use case which require WDT.

Comment (ddsharpe) [reply documentation/1.10/content/userguide/tools/create-aux-image.md:36]: Updated --wdtVersion description.  No, there is no none option included at this time.

Comment (tbarnes-us) [reply documentation/1.10/content/userguide/tools/create-aux-image.md]: MII use case requires WDT yes, but it can have multiple aux images, and only one of them needs to have a WDT install, if any (it could be none of them, since the main image could have the install).

**Review by tbarnes-us — APPROVED** [review]
Approve with caveat:  It's certainly not a showstopper, but I continue to think a "none" option for wdtVersion could be useful (or have a different setting like "--wdtInstall "included|none"). It's easily possible to need an aux image with only model files/archives where the WDT install is already included in a different aux image. (The WKO MII domain home source type supports using multiple aux images in the same domain.)
